tests/driver_nvram_spi/main.c

```diff
@@ -0,0 +1,257 @@
+#include <ctype.h>
+
+#include "board.h"
+#include "vtimer.h"
+#include "periph/spi.h"
+#include "nvram-spi.h"
+
+#ifndef TEST_NVRAM_SPI_DEV
+#error "TEST_NVRAM_SPI_DEV not defined"
+#endif
+#ifndef TEST_NVRAM_SPI_CS
+#error "TEST_NVRAM_SPI_CS not defined"
+#endif
+#ifndef TEST_NVRAM_SPI_SIZE
+#error "TEST_NVRAM_SPI_SIZE not defined"
+#endif
+#ifndef TEST_NVRAM_SPI_ADDRESS_COUNT
+#error "TEST_NVRAM_SPI_ADDRESS_COUNT not defined"
+#endif
+
+#ifdef TEST_NVRAM_SPI_CONF
+#define SPI_CONF    (TEST_NVRAM_SPI_CONF)
+#else
+#define SPI_CONF    (SPI_CONF_FIRST_RISING)
+#endif
+
+#ifdef TEST_NVRAM_SPI_SPEED
+#define SPI_SPEED   (TEST_NVRAM_SPI_SPEED)
+#else
+#define SPI_SPEED   (SPI_SPEED_10MHZ)
+#endif
+
+/* This will only work on small memories. Modify if you need to test NVRAM
+ * memories which do not fit inside free RAM */
+static uint8_t buf_out[TEST_NVRAM_SPI_SIZE];
+static uint8_t buf_in[TEST_NVRAM_SPI_SIZE];
+
+/**
+ * @brief xxd-like printing of a binary buffer
+ */
+static void print_buffer(const uint8_t * buf, size_t length) {
+    static const unsigned int bytes_per_line = 16;
+    static const unsigned int bytes_per_group = 2;
+    unsigned long i = 0;
+    while (i < length) {
+        unsigned int col;
+        for (col = 0; col < bytes_per_line; ++col) {
+            /* Print hex data */
+            if (col == 0) {
+                printf("\n%08lx: ", i);
+            }
+            else if ((col % bytes_per_group) == 0) {
+                putchar(' ');
+            }
+            if ((i + col) < length) {
+                printf("%02hhx", buf[i + col]);
+            } else {
+                putchar(' ');
+                putchar(' ');
+            }
+        }
+        putchar(' ');
+        for (col = 0; col < bytes_per_line; ++col) {
+            if ((i + col) < length) {
+                /* Echo only printable chars */
+                if (isprint(buf[i + col])) {
+                    putchar(buf[i + col]);
+                } else {
+                    putchar('.');
+                }
+            } else {
+                putchar(' ');
+            }
+        }
+        i += bytes_per_line;
+    }
+    /* end with a newline */
+    puts("");
+}
+
+/* weak PRNG for generating "random" test data */
+static uint8_t lcg_rand8(void) {
+    static const uint32_t a = 1103515245;
+    static const uint32_t c = 12345;
+    static uint32_t val = 123456; /* seed value */
+    val = val * a + c;
+    return (val >> 16) & 0xff;
+}
+
+int main(void)
+{
+    uint32_t i;
+    nvram_spi_params_t spi_params = {
+        .spi = TEST_NVRAM_SPI_DEV,
+        .cs = TEST_NVRAM_SPI_CS,
+        .address_count = TEST_NVRAM_SPI_ADDRESS_COUNT,
+    };
+    nvram_t dev;
+    timex_t start_delay = {
+        .seconds = 10,
+        .microseconds = 0,
+    };
+
+    puts("NVRAM SPI test application starting...");
+    printf("Initializing SPI_%i... ", TEST_NVRAM_SPI_DEV);
+    if (spi_init_master(TEST_NVRAM_SPI_DEV, SPI_CONF, SPI_SPEED_10MHZ) == 0) {
```

Comment:
Looking at this example it feels somehow strange to initialzie spi_master manually and after that initializing nvram_spi. This is just a side note